src/com/vivian/java/leetcode/CCI4_1IsBalancedBiTree.java

```diff
@@ -0,0 +1,72 @@
+package com.vivian.java.leetcode;
+
+/* Implement a function to check if a binary tree is balanced.
+
+Claritification: A balanced tree is defined to be a tree that
+the heights of the two subtree of ANY node never differ by more than one
+(-1<= height of rigt - left<=1)
+
+Approach: recursivly determine each node must meet (-1<= height of rigt - left<=1)
+   to make sure left tree is balanced,right tree is balanced
+
+Edgecase: only determine the height difference is not enough,
+         for example: we can have its subtree is balanced, but its left subtree isn't balanced
+
+Complexity: runtime complexity o(n^2),One each node, we recurse its entire subTree(means one each node,we called height() repeatly)
+
+*/
+public class CCI4_1IsBalancedBiTree {
+    public boolean isBlanced(BiTreeNode<Integer> root) {
+        if(root == null) {
+            return true;
+        }
+        int left_height = height(root.getLeftNode());
+        int right_height = height(root.getRightNode());
+        if(Math.abs(left_height - right_height) > 1) {
+            return false;
+        } else {
+            return isBlanced(root.getRightNode()) && isBlanced(root.getLeftNode());
+        }
+    }
+
+    public int height(BiTreeNode<Integer> root) {
+        if(root == null) {
+            return 0;
+        }
+        return Math.max(height(root.getLeftNode()), height(root.getRightNode())) + 1;
+    }
+}
+
+/*
```

Comment:
No need to comment approach 2 out, and you should test both approaches.

Reply:
tested